test/unit/spec/shared_examples.rb

```diff
@@ -0,0 +1,127 @@
+shared_examples 'standard install' do |platform, version, package, expected_url|
+  context "with the '#{package}' package" do
+    let(:package) { package }
+    let(:package_path) { "./test/unit/.cache/#{filename_from_url(expected_url)}" }
+    let(:runner_params) { { platform: platform.to_s, version: version.to_s } }
+    let(:test_params) { { name: package.to_s, build: 'cae2458f4aef', version: '6.3.4' } }
+
+    it 'should set the run state' do
+      expect(run_state['current_installation']).to eq(
+        name: package.to_s,
+        package: package,
+        version: '6.3.4',
+        build: 'cae2458f4aef',
+        x64: true
+      )
+    end
+
+    it 'should download the package' do
+      expect(chef_run).to create_remote_file(package_path).with(source: expected_url)
+    end
+
+    case platform
+    when :redhat
+      it 'should install from the rpm package' do
+        expect(chef_run).to install_rpm_package(package_name).with(source: package_path)
+      end
+    when :ubuntu
+      it 'should install from the dpkg package' do
+        expect(chef_run).to install_dpkg_package(package_name).with(source: package_path)
+      end
+    when :windows
+      it 'should install from the msi package' do
+        expect(chef_run).to install_windows_package(package_name).with(
+          options: 'LAUNCHSPLUNK=0 INSTALL_SHORTCUT=0 AGREETOLICENSE=Yes'
+        )
+      end
+    else
+      it 'should install from the tgz package' do
+        expect(chef_run).to extract_local_tar_extract(package_path).with(target_dir: install_dir)
+      end
+    end
+
+    context "when '#{package}' is already installed" do
```

Comment:
can you add a context for when the package is already installed but with a different version?